lib/evse_security/evse_security.cpp

```diff
@@ -1708,15 +1709,16 @@ CertificateValidationResult EvseSecurity::verify_certificate_internal(const std:
         const auto leaf_certificate = _certificate_chain.at(0);
 
         // Retrieve the hierarchy in order to check if the chain contains a root certificate
-        X509CertificateHierarchy& hierarchy = certificate.get_certficate_hierarchy();
+        X509CertificateHierarchy& hierarchy = certificate.get_certificate_hierarchy();
 
         // Build all untrusted intermediary certificates, and exclude any root
         std::vector<X509Handle*> untrusted_subcas;
 
         if (_certificate_chain.size() > 1) {
             for (size_t i = 1; i < _certificate_chain.size(); i++) {
                 const auto& cert = _certificate_chain[i];
-                if (hierarchy.is_root(cert)) {
+                // Ignore our root certificates or if the received certificate is somehow self-signed
+                if (cert.is_selfsigned() || hierarchy.is_internal_root(cert)) {
```

Comment:
Isnt `cert.is_selfsigned()` a sufficient check? If it is not self signed and can not be an internal root anyways, right?